src/wp-includes/user.php

```diff
@@ -604,9 +604,19 @@ function wp_validate_logged_in_cookie( $user_id ) {
 function count_user_posts( $userid, $post_type = 'post', $public_only = false ) {
 	global $wpdb;
 
+	$post_type = (array) $post_type;
```

Comment:
```suggestion
	$post_type = array_filter( array_unique( (array) $post_type ), 'post_type_exists' );
```
@peterwilsoncc What do you think of this? Overkill?

Reply:
`array_unique()` would help, `get_posts_by_author_sql()` already ensures the post type exists so no need to do it here.

Reply:
Well that's the point, this code would result in hitting the cache, as the cache key would match. 

So 
```
count_user_posts( 1, array( 'post', 'invalid' ) );
```
Or 
```
count_user_posts( 1, array( 'post', '' ) );
```
Would be the same as 

```
count_user_posts( 1, array( 'post') );
```

Reply:
That's what I am saying, they should already hit the cache as the it's generated based on the SQL without taking in to account the arguments passed to the function.